src/stdlib_io.fypp

```diff
@@ -331,4 +340,59 @@ contains
 
   end function parse_mode
 
+  !> Version: experimental
+  !>
+  !> Read a whole line from a formatted unit into a deferred length character variable
+  subroutine getline_char(unit, line, iostat, iomsg)
+    !> Formatted IO unit
+    integer, intent(in) :: unit
+    !> Line to read
+    character(len=:), allocatable, intent(out) :: line
+    !> Status of operation
+    integer, intent(out), optional :: iostat
+    !> Error message
+    character(len=:), allocatable, optional :: iomsg
+
+    integer, parameter :: bufsize = 512
```

Comment:
512 seems reasonable. Is there any rule of thumb on the best value for this? I assume not, as it depends on what data the users will read. If it later shows that it's important, we could make it user settable by making `bufsize` an optional argument and `buffer` an allocatable.

Reply:
Not sure, best is something that captures most usual line lengths at once. The actual buffer length should be an implementation detail and not concern the user or clutter the API in my opinion.

Reply:
```suggestion
    integer, parameter :: bufsize = 4096
```